it/uk/gov/hmrc/transitmovementsrouter/connectors/PersistenceConnectorSpec.scala

```diff
@@ -205,4 +228,36 @@ class PersistenceConnectorSpec
     }
 
   }
+
+  "patch for update message status" should {
+    "return 200 when it is successful" in {
+
+      stubForPatchMessageStatus(OK)
+
+      whenReady(connector.patchMessageStatus(movementId, messageId, MessageUpdate(MessageStatus.Success)).value) {
+        x =>
+          x.isRight mustBe true
+          x mustBe Right(())
+      }
+    }
+
+    "return a PersistenceError when unsuccessful" in forAll(errorCodes) {
+      statusCode =>
+        server.resetAll()
+
+        stubForPatchMessageStatus(statusCode)
+
+        whenReady(connector.patchMessageStatus(movementId, messageId, MessageUpdate(MessageStatus.Success)).value) {
+          x =>
+            x.isLeft mustBe true
+
+            statusCode match {
+              case BAD_REQUEST           => x mustBe a[Left[Unexpected, _]]
+              case NOT_FOUND             => x mustBe a[Left[MovementNotFound, _]]
+              case INTERNAL_SERVER_ERROR => x mustBe a[Left[Unexpected, _]]
```

Comment:
So this is interesting, I totally get what you've done here but thanks to type erasure on the JVM, this is always going to work because it'll collapse to checking if the object is just a left.

Try this:

```scala
whenReady(connector.patchMessageStatus(movementId, messageId, MessageUpdate(MessageStatus.Success)).value) {
    case Left(error) =>
        statusCode match {
              case BAD_REQUEST           => x mustBe a[Unexpected]
              case NOT_FOUND             => x mustBe a[MovementNotFound]
              case INTERNAL_SERVER_ERROR => x mustBe a[Unexpected]
        }
    case Right(_) => fail("An error was expected")
}
```

That should avoid type erasure issues.

Reply:
sure, updated